pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -278,11 +278,40 @@ static int is_simple_frame(PyObject *obj) {
 }
 
 static npy_int64 get_long_attr(PyObject *o, const char *attr) {
+    // NB we are implicitly assuming that o is a Timedelta or Timestamp, or NaT
+
     npy_int64 long_val;
     PyObject *value = PyObject_GetAttrString(o, attr);
     long_val =
         (PyLong_Check(value) ? PyLong_AsLongLong(value) : PyLong_AsLong(value));
+
     Py_DECREF(value);
+
+    if (long_val == NPY_MIN_INT64) {
+        // i.e. o is NaT
+        return long_val;
+    }
+
+    // ensure we are in nanoseconds, similar to Timestamp._as_creso or _as_unit
+    PyObject* reso = PyObject_GetAttrString(o, "_creso");
+    // if (!PyLong_Check(reso)) {
+    // TODO(anyone): we should have error handling here, but one step at a time
+    // }
+
+    long cReso = PyLong_AsLong(reso);
+    // if (cReso == -1 && PyErr_Occurred()) {
+    // TODO(anyone): we should have error handling here, but one step at a time
+    // }
+    Py_DECREF(reso);
```

Comment:
Should move this before the above check

Reply:
should this also occur inside the short-circuiting return on reso validation?

Reply:
Nope - you only need to DECREF reso one time. If you DECREF again in the return you would be double freeing.

The assignment to `cReso` copies the value that you want, so reso can be DECREFed immediately thereafter whether things failed or not

Reply:
i mean before L299.  if we go reach L299, won't this Py_DECREF not get hit?

Reply:
Ah good catch - yea you are correct